### ISSUE TITLE
[BUG] bokeh authentication example checks only whether `user` cookie is set

### ISSUE THREAD

**Issue by ltalirz** [bokeh/bokeh#10463]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
bokeh `master` branch

#### Observed behavior
I'm trying out the authentication features for the first time, so I looked at the corresponding [howto example](https://github.com/bokeh/bokeh/tree/main/examples) in the bokeh repository.

Without a complete understanding of the inner workings, it seems to me the first thing that bokeh does is call the `get_user` function
https://github.com/bokeh/bokeh/blob/b5abc5a5b32b283b3b3cdaebf430728a06d84aba/examples/howto/server_auth/auth.py#L5-L7
which checks the `user` cookie.

It turns out that if that function returns anything but `None`, the `LoginHandler` is never activated (!), i.e. the user is assumed to be already logged in.
I.e. setting the `user` cookie manually allows any user to circumvent the user/password check implemented here:
https://github.com/bokeh/bokeh/blob/b5abc5a5b32b283b3b3cdaebf430728a06d84aba/examples/howto/server_auth/auth.py#L22-L25

Maybe this is obvious to people who are familiar with the details of the authentication workflow, but this code is likely to be copied by people who are new to this (like me), who may be surprised to find that the apps they thought are login-protected really aren't.

#### Expected behavior
I would expect that I cannot access the protected app without a valid user name and password.

#### Complete, minimal, self-contained example code that reproduces the issue
Check out the bokeh repository and do:
```
cd examples/howto/server_auth
bokeh serve --enable-xsrf-cookies --auth-module=auth.py  app.py
```

Open http://localhost:5006/app without logging in.

Open the browser console and execute 
```
document.cookie="user=test"
```

Visit http://localhost:5006/app again - you are logged in!

**Comment by ltalirz** [comment]
P.S. If others agree that this should be fixed, I'm happy to make a PR for any suggested fix.

**Comment by bryevdv** [comment]
This is intentionally a toy example that is only meant to minimally illustrate how the building blocks fit together. It's definitely incumbent on anyone using these to build out a real auth workflow that matches their unique circumstances. The only change I would personally advocate is some loud comments to that effect. 

cc @bokeh/dev

**Comment by bryevdv** [comment]
I should clarify that adding warning comments is the only change I would make *to the example itself*. We can always add more supporting documentation, and certainly a discussion of common workflows would be welcome. @philippjfr haven't you worked with folks that implemented an OAuth workflow? Can you summarize the experience?

BTW the problem with making the example "real" is that it would require a real authorization server to be configured and running somewhere. Given that, I'd much rather make sure that it be *very extremely obviously a toy demonstration* rather than "improving" it in any way that that makes it seem plausibly complete. But additional viewpoints would be appreciated.

**Comment by Khalilsqu** [comment]
if this a toy that should not be used for production, where someone like me who does not understand much of programming and thousands of web frameworks can find production-ready examples? bokeh or Panel does not provide many examples on how to use their framework, and sorry to say that but their documentation is not good either. 

Streamlit has amazing documentation and many examples on how to use each feature can easily be found. this is the real reason streamlit sky-rocketed in popularity.

**Comment by philippjfr** [comment]
>Streamlit has amazing documentation and many examples on how to use each feature can easily be found. this is the real reason streamlit sky-rocketed in popularity.

Streamlit also happens to be a company with tens of millions in VC funding which has since been acquired by one of the largest companies in the world while Bokeh and Panel are largely run by volunteers with some backing by our employers. With limited resources we rely on contributors to improve the libraries.

If you have suggestions on improving the [OAuth documentation in Panel](https://panel.holoviz.org/user_guide/Authentication.html) I will **very happily** hear them and would be even happier about a PR.

**Comment by bryevdv** [comment]
@Khalilsqu it's users like youv who make OSS developers consider quitting open source altogether. Please know that your comment was not useful, valuable, or helpful in any way.

**Comment by bryevdv** [comment]
I've opened #12516 which emphasizes the intended purpose and audience for this example to every file in `server_auth`. I hope that is sufficient.
